server/build/webpack.js

```diff
@@ -33,17 +33,13 @@ function babelConfig (dir, {isServer, dev}) {
   }
 
   const externalBabelConfig = findBabelConfig(dir)
-  if (externalBabelConfig) {
+  if (externalBabelConfig && externalBabelConfig.babelrc) {
     // Log it out once
     if (!isServer) {
       console.log(`> Using external babel configuration`)
-      console.log(`> Location: "${externalBabelConfig.loc}"`)
+      console.log(`> Location: "${externalBabelConfig.babelrc}"`)
     }
-    // It's possible to turn off babelrc support via babelrc itself.
-    // In that case, we should add our default preset.
-    // That's why we need to do this.
-    const { options } = externalBabelConfig
-    mainBabelOptions.babelrc = options.babelrc !== false
+    mainBabelOptions.babelrc = true
```

Comment:
If you change up this logic to just be per-filename like I'd recommend, you shouldn't need to assign this to anything. e.g.

```
const presetItem = createConfigItem(require.resolve('./babel/preset'));
const hotLoaderItem = createConfigItem(require.resolve('react-hot-loader/babel'));

function getBabelConfig(filename, {isServer, dev}) {
  const config = loadPartialConfig({ 
    filename,
    plugins: [
      dev && !isServer && hotLoaderItem
    ].filter(Boolean),
  });

  if (!config) return null;

  let options = config.options;
  if (config.babelrc) {
    if (!isServer) {
      // Maybe just have some extra logic to avoid printing this?
      console.log(`> Using external babel configuration`)
      console.log(`> Location: "${config.babelrc}"`)
    }
  } else {
    options.presets.push(presetItem);
  }

  return options;
}
```

Reply:
Hmmm in that case babel-loader needs to be changed to support a function that has the file path as argument and resolves to a babel config right 🤔

Reply:
I like the idea though 👍

Reply:
Ahh, I'd totally forgotten that this is used via babel-loader. That makes this a lot clearer.

> Hmmm in that case babel-loader needs to be changed to support a function that has the file path as argument and resolves to a babel config right

Maybe that's something we should explore. I hadn't thought about it but it may well make sense.

Reply:
Unrelated to this specific comment, I'd remove the `mainBabelOptions.babelrc = true` line, since searching for the config file is already the default.

Reply:
I've removed the line 👍

Reply:
@loganfsmyth created an issue here: https://github.com/babel/babel-loader/issues/598